### ISSUE TITLE
[java] LawOfDemeter: False positive for chained methods with generic method call

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2175]
https://github.com/pmd/pmd/issues/2160#issuecomment-565639733

**Affects PMD Version:**
6.20.0

**Rule:**
[LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter)

**Description:**
The rule double counts dot accessor for method that uses angled brackets to specify return type in generic method.

**Code Sample demonstrating the issue:**

```java
package pmdtests;

import java.util.Arrays;
import java.util.List;

public enum Containment {
  ;

  public static void main(String... args) {
    Object obj = "one";
    List<Object> objs = Arrays.asList("one", 2, Math.PI, 4);
    List<Integer> ints = Arrays.asList(2, 4);
    assert objs.contains(obj);     // wrong violation: object not created locally
    assert objs.containsAll(ints); // wrong violation: object not created locally
    assert !ints.contains(obj);    // wrong violation: object not created locally
    assert !ints.containsAll(objs);// wrong violation: object not created locally

    obj = 1;
    objs = Arrays.<Object>asList(1, 3); // wrong violation: method chain calls
    ints = Arrays.asList(1, 2, 3, 4);
    assert ints.contains(obj);          // wrong violation: object not created locally
    assert ints.containsAll(objs);      // wrong violation: object not created locally
  }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
